### PR TITLE
Topic/cmdline

### PR DESCRIPTION
**Pull request by godunko**

### CODE REVIEW

File: source/os/vss-command_line.ads [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   type Name_Value_Option is new Named_Option with record
+      Name_Name  : VSS.Strings.Virtual_String;
+      Value_Name : VSS.Strings.Virtual_String;
+   end record;
+
+   procedure Add_Option (Option : Abstract_Option'Class);
+
+   procedure Process;
+   --  Parse command line. If error is found then report it and terminate
+   --  application.
```
Comment (reznikmm) [inline source/os/vss-command_line.ads:52]: Terminating application looks as a bad idea to me. I prefer this procedure to return a parsing status and let the user react by her self.

Comment (godunko) [reply source/os/vss-command_line.ads:52]: It is part of contract for this subprogram, it makes application code cleaner. 

Subprogram to do parsing with status may be added later if necessary.

File: source/os/implementation/vss-command_line-platform__windows.adb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+with VSS.Implementation.Windows.Kernel32;
+with VSS.Implementation.Windows.String_Utilities;
+with VSS.Implementation.Windows.User32;
+
+separate (VSS.Command_Line)
+package body Platform is
+
+   function Use_Message_Box return Boolean;
+   --  Return True then message bosy should be used instead of standard error
```
Comment (AnthonyLeonardoGracio) [inline source/os/implementation/vss-command_line-platform__windows.adb:17]: "then message box"

**Review by reznikmm — APPROVED** [review]

File: source/os/implementation/vss-command_line-platform__windows.adb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+with Interfaces.C;
+
+with VSS.Implementation.Windows.Kernel32;
+with VSS.Implementation.Windows.String_Utilities;
+with VSS.Implementation.Windows.User32;
+
+separate (VSS.Command_Line)
+package body Platform is
+
+   function Use_Message_Box return Boolean;
```
[the remark this answers is not in the document]
Comment (godunko) [reply source/os/implementation/vss-command_line-platform__windows.adb:16]: Message box is used when application is run directly by user. Scripts redirect standard streams, this is detected by this code and stderr is used for the output.

**Pull request merged**